apps/deploy-web/src/components/new-deployment/NewDeploymentContainer.tsx

```diff
@@ -19,7 +19,7 @@ import { TemplateList } from "./TemplateList";
 export function NewDeploymentContainer() {
   const { isLoading: isLoadingTemplates, templates } = useTemplates();
   const [activeStep, setActiveStep] = useState<number | null>(null);
-  const [selectedTemplate, setSelectedTemplate] = useState<TemplateCreation | null>(null);
+  const [selectedTemplate, onTemplateSelected] = useState<TemplateCreation | null>(null);
```

Comment:
Any particular reason you renamed this to `onTemplateSelected`? Usual convention for naming `useState` setters is `set...` and `on..` is more for event handler functions.

Reply:
It could be like this:
```
const onTemplateSelected = (template: string) => {
  setSelectedTemplate(template);
}

...

onTemplateSelected={onTemplateSelected}
```

Reply:
Yes, right.

Reply:
guys, this is wrong. You have a method in ur component which should say what it does. When you name it `onTemplateSelected` you introduce confusion as actually you don't know you just know when it happens. So you go and read the method code. However this is a good name to expose for a callback, because you want the client to react when something happens and you don't care about external logic. So overall this should simply be:


```typescript
const [selectedTemplate, setSelectedTemplate] = useState<TemplateCreation | null>(null);

onTemplateSelected={setSelectedTemplate}
```

Reply:
@ygrishajev @baktun14 
Your suggestions all make sense.

If we foresee the need to add any additional logic, the @baktun14's approach is preferable for its flexibility and clarity.
However, if simplicity and conciseness are paramount and we are certain that no additional logic will be needed, the @ygrishajev's approach is valid and keeps the code minimal and straightforward.

Reply:
At this moment we don't need to add any additional logic to the onTemplateSelected function,
I suggest @ygrishajev's approach.

Reply:
I agree with ur summary @hiroyukikumazawa. This thread is resolved for me.

What I meant here although is naming approaches with a custom function or without it. I don't see a need to for a fn calling another single function though indeed.